pkg/machines/libvirt-dbus.js

```diff
@@ -1478,6 +1483,34 @@ export function networkUndefine(connectionName, objPath) {
     return call(connectionName, objPath, 'org.libvirt.Network', 'Undefine', [], TIMEOUT);
 }
 
+export function setOSFirmware(connectionName, objPath, loaderType) {
+    return call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], TIMEOUT)
+            .then(domXml => {
+                const domainElem = getDomainElem(domXml);
+
+                if (!domainElem)
+                    throw new Error("setOSFirmware: domXML has no domain element");
+
+                const osElem = domainElem.getElementsByTagNameNS("", "os")[0];
+                const loaderElem = getSingleOptionalElem(osElem, "loader");
+
+                if (loaderElem)
+                    loaderElem.remove();
```

Comment:
loader element ans os->firware attributes can be used together. User can have "secure" option specified in it's loader element, which can be used together with os->firmware, so with this change they could loose that secure configuration if we just delete loaderElem.

Reply:
When we are setting the firmware from the UI we will always set the firmware attribute to `efi` or `bios`. All the rest will be autofilled.
It doesn't make sense to keep existing loader elements  since they might not be in sync with the firmware attribute.
In addition a user will  only have the loader element set if they configured it through CLI, virt-manager or some other tool. So they probably know what they are doing and they should not expect that their settings will be kept if they try to re-configure it through Cockpit UI.

So all in all, I think we are ok with keeping this one.